packages/relay-service/lib/index.js

```diff
@@ -80,19 +81,31 @@ const getMessages = async ({keyPairDocs, limit = 20}) => {
         const keyAgreementKey = await getDerivedAgreementKey(keyPairDoc);
         let jwe = item.msg;
 
-        if (isBase64(jwe)) {
-          jwe = fromBase64(jwe);
+        if (typeof jwe === 'string') {
+          // TODO: check for JWT in future here
+          try {
+            if (isBase64(jwe)) {
+              jwe = fromBase64(jwe);
+            } else {
+              jwe = JSON.parse(jwe);
+            }
+          } catch (e) {
+            Logger.debug(`Invalid JWE message received: ${jwe}`);
+            console.error(e);
+            return null;
+          }
         }
 
         const didCommMessage = await didcomm.decrypt(jwe, keyAgreementKey);
         return {
```

Comment:
```
        return {
          ...item,
          msg: didCommMessage,
        };
```

is better to prevent duplication, especially for large messages

Reply:
fixed